tools/src/main/scala/scala/scalanative/linker/ClassLoader.scala

```diff
@@ -41,6 +47,18 @@ object ClassLoader {
       out
     }
 
+    def classesWithEntryPoints(): Iterable[Global] = {
+      scopes.filter {
+        case (top, defns) =>
+          defns.exists {
+            case defn: Defn.Define =>
+              val Global.Member(_, sig) = defn.name
+              sig.isClinit
+            case _ => false
+          }
+      }.keySet
```

Comment:
Even in memory, I believe we should cache this result, because it is expensive to compute, and used on every linking request.